### PR TITLE
Add HTML comments

### PR DESCRIPTION
**Pull request by nmote**
This is part of the ES Spec (https://tc39.es/ecma262/#sec-html-like-comments), and although it practically begs you not to implement it, it is valid and browsers are required to handle it. I've given examples in the associated test and have given an explanation inline in the grammar.

Checklist:
- [x] All tests pass in CI.
- [x] The script/parse-examples passes without issues.
- [x] There are sufficient tests for the new fix/feature.
- [x] Grammar rules have not been renamed unless absolutely necessary.
- [x] The conflicts section hasn't grown too much.
- [x] The parser size hasn't grown too much (check the value of STATE_COUNT in src/parser.c).

### CODE REVIEW

**Review by brandonspark — APPROVED** [review]

**Comment by nmote** [comment]
Waiting for approval for CI to check the first box.

**Review by aryx — APPROVED** [review]

**Pull request merged**
